modules/ci_environment/manifests/transition_logs/s3_backups.pp

```diff
@@ -0,0 +1,42 @@
+# class ci_environment::transition_logs::s3_backups
+#
+# [*directory*]
+#   Defines the directory to backup
+#
+# [*hour*]
+#   Defines the hour to run the cron job
+#
+# [*minute*]
+#   Defines the minute to run the cron job
+#
+# [*s3_bucket*]
+#   Defines the AWS S3 bucket destination for backups
+#
+# [*s3_aws_id*]
+#   Defines the AWS access id with permission to write to 
+#   the S3 bucket
+#
+# [*s3_aws_key*]
+#   Defines the AWS access secret key for the defined aws access id
+
+class ci_environment::transition_logs::s3_backups(
+  $directory = '/srv/logs/log-1',
+  $hour = 4,
+  $minute = 15,
+  $s3_bucket = undef,
+  $s3_aws_id = undef,
+  $s3_aws_key = undef,
+  ){
+
+  duplicity { 'transition-logs-s3-backups':
+    directory         => $directory,
+    bucket            => $s3_bucket,
+    dest_id           => $s3_aws_id,
+    dest_key          => $s3_aws_key,
+    pubkey_id         => '13B84C37AB52D76B3F53CF0E7C34BD7A05119BA4',
+    hour              => $hour,
+    minute            => $minute,
+    remove_older_than => '30D',
```

Comment:
I've tried to run this pull request in vagrant and it turns out the documentation for puppet-duplicity is out-of-date. I've opened alphagov/puppet-duplicity#10 to fix it. The problem is that `remove_older_than` doesn't exist anymore. The error when spinning up this pull request in vagrant is:

```
==> transition-logs-1: Error: Invalid parameter remove_older_than on Duplicity[transition-logs-s3-backups] at /tmp/vagrant-puppet/modules-a35c1534eadee100a9485aa8828328fe/ci_environment/manifests/transition_logs/s3_backups.pp:40 on node transition-logs-1.internal
```